pyface/ui/wx/util/image_helpers.py

```diff
@@ -0,0 +1,177 @@
+    smooth = wx.IMAGE_QUALITY_HIGH
+
+
+class AspectRatio:
+    ignore = 0
+    keep_constrain = 1
+    keep_expand = 2
+
+
+def image_to_bitmap(image):
+    """ Convert a wx.Image to a wx.Bitmap.
+    
+    Parameters
+    ----------
+    image : wx.Image
+        The wx.Image to convert.
+
+    Return
+    ------
+    bitmap : wx.Bitmap
+        The corresponding wx.Bitmap.
+    """
+    return image.ConvertToBitmap()
+
+
+def bitmap_to_image(bitmap):
+    """ Convert a wx.Bitmap to a wx.Image.
+    
+    Parameters
+    ----------
+    bitmap : wx.Bitmap
+        The wx.Bitmap to convert.
+
+    Return
+    ------
+    image : wx.Image
+        The corresponding wx.Image.
+    """
+    return bitmap.ConvertToImage()
+
+
+def bitmap_to_icon(bitmap):
+    """ Convert a wx.Bitmap to a wx.Icon.
+        
+    Parameters
+    ----------
+    bitmap : wx.Bitmap
+        The wx.Bitmap to convert.
+
+    Return
+    ------
+    icon : wx.Icon
+        The corresponding wx.Icon.
+    """
+    return wx.Icon(bitmap)
+
+
+def resize_image(image, size, aspect_ratio=AspectRatio.ignore,
+                 mode=ScaleMode.fast):
+    """ Resize a toolkit image to the given size. """
+    image_size = image.GetSize()
+    width, height = _get_size_for_aspect_ratio(image_size, size, aspect_ratio)
+    return image.Scale(width, height, mode)
+
+
+def resize_bitmap(bitmap, size, aspect_ratio=AspectRatio.ignore,
+                  mode=ScaleMode.fast):
+    """ Resize a toolkit bitmap to the given size. """
+    image = bitmap_to_image(bitmap)
+    image = resize_image(image, size, aspect_ratio, mode)
+    return image_to_bitmap(image)
+
+
+def image_to_array(image):
+    """ Convert a wx.Image to a numpy array.
+
+    This copies the data returned from Qt.
+
+    Parameters
+    ----------
+    image : wx.Image
+        The wx.Image that we want to extract the values from.  The format must
+        be either RGB32 or ARGB32.
+
+    Return
+    ------
+    array : ndarray
+        An N x M x 4 array of unsigned 8-bit ints as RGBA values.
+    """
+    import numpy as np
+
+    width, height = image.GetSize()
+    rgb_data = np.array(image.GetData(), dtype='uint8').reshape(width, height, 3)
+    if image.HasAlpha():
+        alpha = np.array(image.GetAlpha(), dtype='uint8').reshape(width, height)
+    else:
+        alpha = np.full((width, height), 0xff, dtype='uint8')
+    array = np.empty(shape=(width, height, 4), dtype='uint8')
+    array[:, :, :3] = rgb_data
+    array[:, :, 3] = alpha
+    return array
+
+
+def array_to_image(array):
+    """ Convert a numpy array to a wx.Image.
+
+    This copies the data before passing it to Qt.
+
+    Parameters
+    ----------
+    array : ndarray
+        An N x M x {3, 4} array of unsigned 8-bit ints.  The image
+        format is assumed to be RGB or RGBA, based on the shape.
+
+    Return
+    ------
+    image : wx.Image
+        The wx.Image created from the data.  The pixel format is
+        wx.Image.Format_RGB32.
```

Comment:
it looks like this function might return an image with 4 channels so shouldn't this be `RGB32 or ARGB32`?

Reply:
Actually this is a search and replace bug.  `wx.Image` doesn't have a notion of format, it's all "RGB" with an optional seperate alpha channel:
```suggestion
        The wx.Image created from the data.
```